tests/algorithms/test_algorithm_resumption.py

```diff
@@ -0,0 +1,236 @@
+])
+@pytest.mark.parametrize("device_hparams,deepspeed_enabled,zero_stage", [
+    pytest.param(CPUDeviceHparams(), False, None, id="cpu-ddp"),
+    pytest.param(GPUDeviceHparams(), False, None, id="gpu-ddp", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 0, id="deepspeed-zero0", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 1, id="deepspeed-zero1", marks=pytest.mark.gpu),
+    pytest.param(GPUDeviceHparams(), True, 2, id="deepspeed-zero2", marks=pytest.mark.gpu),
+])
+@pytest.mark.parametrize(
+    "seed,save_interval,save_name_format,resume_file,final_checkpoint",
+    [
+        [None, "1ep", "ep{epoch}", "ep3", "latest-rank{rank}"],  # test randomized seed saving and symlinking
+        [42, "1ep", "ep{epoch}", "ep3", "ep5"],  # test save at epoch end
+        [42, "1ep", "ep{epoch}.tgz", "ep3.tgz", "ep5.tgz"],  # test tarball with compression
+        [42, "2ba", "ba{batch}", "ba12", "ba25"],  # test save batch in partial epoch
+        [42, "1ba", "ba{batch}", "ba15", "ba25"],  # test save final batch in epoch
+        [42, "2ba", "ba{batch}", "ba16", "ba25"],  # test save batch after complete epoch
+    ],
+)
+@pytest.mark.parametrize("model_name", [None, "resnet50_synthetic", "gpt2_52m"])
+@pytest.mark.parametrize("algorithm", algorithms_registry.items(), ids=algorithms_registry.keys())
+def test_algorithm_resumption(
+    device_hparams: DeviceHparams,
+    world_size: int,
+    deepspeed_enabled: bool,
+    zero_stage: Optional[int],
+    composer_trainer_hparams: TrainerHparams,
+    save_interval: str,
+    save_name_format: str,
+    resume_file: str,
+    final_checkpoint: str,
+    seed: Optional[int],
+    model_name: Optional[str],
+    algorithm: Tuple[str, Type[AlgorithmHparams]],
+):
+    """strategy:
+    - train five epochs. capture checkpoints after `checkpoint_interval` and ep5.
+    - create a new trainer from the `checkpoint_interval` checkpoint, and train until end. checkpoint again.
+    - assert that the checkpoint from the new trainer at the end is the same as the checkpoint from the first trainer at the end.
+    """
+    del world_size  # unused. Read via env variable
+
+    algo_name, algo_hparams = algorithm
+
+    if not isinstance(device_hparams, GPUDeviceHparams) and deepspeed_enabled:
+        pytest.skip("DeepSpeed tests must be ran on GPU")
+
+    if deepspeed_enabled:
+        if not _is_archive(resume_file):
+            resume_file += ".tar"
+        if not _is_archive(final_checkpoint):
+            final_checkpoint += ".tar"
+
+    composer_trainer_hparams.schedulers = []
+    composer_trainer_hparams.max_duration = '5ep'
+    if model_name in ["resnet50_synthetic", "gpt2_52m"]:
+        if not isinstance(device_hparams, GPUDeviceHparams):
+            pytest.skip("Real models require a GPU -- otherwise they take too long")
+        model_hparams = TrainerHparams.load(model_name)
+        composer_trainer_hparams.train_dataset = model_hparams.train_dataset
```

Comment:
ah i can see why it's easier to use the trainer hparams objects now (please disregard my previous comment). A few suggestions:
* since we basically overwrite all the defaults from the `composer_trainer_hparams` object, can you just create it directly here? e.g.
```
hparams = TrainerHparams(...)
```
this change would help readability without having to refer to the fixture file.

* use a much shorter variable name (e.g. `trainer_hparams`, or just `hparams`).